pkg/operator/ceph/object/objectstore.go

```diff
@@ -765,17 +780,235 @@ func CreatePools(context *Context, clusterSpec *cephv1.ClusterSpec, metadataPool
+		if pool.Name == sharedPools.MetadataPoolName {
+			foundMetadataPool = true
+		}
+		if pool.Name == sharedPools.DataPoolName {
+			foundDataPool = true
+		}
+	}
+
+	if !foundMetadataPool && !foundDataPool {
+		return fmt.Errorf("pools do not exist: %q and %q", sharedPools.MetadataPoolName, sharedPools.DataPoolName)
+	}
+	if !foundMetadataPool {
+		return fmt.Errorf("metadata pool does not exist: %q", sharedPools.MetadataPoolName)
+	}
+	if !foundDataPool {
+		return fmt.Errorf("data pool does not exist: %q", sharedPools.DataPoolName)
+	}
+
+	logger.Info("verified shared pools exist")
+	return nil
+}
+
+func applyExpectedRadosNamespaceSettings(zoneConfig map[string]interface{}, metadataPrefix, dataPrefix, dataPool string) error {
+	// Update the necessary fields for RAODS namespaces
+	zoneConfig["domain_root"] = metadataPrefix + "meta.root"
+	zoneConfig["control_pool"] = metadataPrefix + "control"
+	zoneConfig["gc_pool"] = metadataPrefix + "log.gc"
+	zoneConfig["lc_pool"] = metadataPrefix + "log.lc"
+	zoneConfig["log_pool"] = metadataPrefix + "log"
+	zoneConfig["intent_log_pool"] = metadataPrefix + "log.intent"
+	zoneConfig["usage_log_pool"] = metadataPrefix + "log.usage"
+	zoneConfig["roles_pool"] = metadataPrefix + "meta.roles"
+	zoneConfig["reshard_pool"] = metadataPrefix + "log.reshard"
+	zoneConfig["user_keys_pool"] = metadataPrefix + "meta.users.keys"
+	zoneConfig["user_email_pool"] = metadataPrefix + "meta.users.email"
+	zoneConfig["user_swift_pool"] = metadataPrefix + "meta.users.swift"
+	zoneConfig["user_uid_pool"] = metadataPrefix + "meta.users.uid"
+	zoneConfig["otp_pool"] = metadataPrefix + "otp"
+	zoneConfig["notif_pool"] = metadataPrefix + "log.notif"
+
+	placementPools, ok := zoneConfig["placement_pools"].([]interface{})
+	if !ok {
+		return fmt.Errorf("failed to parse placement_pools")
+	}
+	if len(placementPools) == 0 {
+		return fmt.Errorf("no placement pools")
+	}
+	placementPool, ok := placementPools[0].(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("failed to parse placement_pools[0]")
+	}
+	placementVals, ok := placementPool["val"].(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("failed to parse placement_pools[0].val")
+	}
+	placementVals["index_pool"] = metadataPrefix + "buckets.index"
+	placementVals["data_extra_pool"] = dataPrefix + "buckets.non-ec"
+	storageClasses, ok := placementVals["storage_classes"].(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("failed to parse storage_classes")
+	}
+	stdStorageClass, ok := storageClasses["STANDARD"].(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("failed to parse storage_classes.STANDARD")
+	}
+	stdStorageClass["data_pool"] = dataPool
+	return nil
+}
+
+func dataPoolIsExpected(objContext *Context, zoneConfig map[string]interface{}, expectedDataPool string) bool {
+	placementPools, ok := zoneConfig["placement_pools"].([]interface{})
+	if !ok {
+		return false
+	}
+	placementPool, ok := placementPools[0].(map[string]interface{})
```

Comment:
is this we are checking for expected values, why not also use xvalidation for this?

Reply:
xvalidation is on the crds, but this data pool config was returned by a ceph query. If the schema changes unexpectedly from ceph, we expect to detect the failure in the CI and then will need a code change. There is no ceph api to retrieve this info though.